src/console_commands.cpp

```diff
@@ -51,58 +51,123 @@ ConsoleCommands::~ConsoleCommands()
 	LOGINFO(1, "stopped");
 }
 
+typedef struct strconst {
+	const char *str;
+	size_t len;
+} strconst;
+
+#define STRCONST(x)	{x, sizeof(x)-1}
+#define STRCNULL	{NULL, 0}
+
+typedef int (cmdfunc)(p2pool *pool, const char *args);
+
+typedef struct cmd {
+	strconst name;
+	const char *arg;
+	const char *descr;
+	cmdfunc *func;
+} cmd;
+
+static cmdfunc do_help, do_status, do_loglevel, do_addpeers, do_droppeers, do_exit;
+
+static cmd cmds[] = {
+	{ STRCONST("help"), "", "display list of commands", do_help },
+	{ STRCONST("status"), "", "display p2pool status", do_status },
+	{ STRCONST("loglevel"), "<level>", "set log level", do_loglevel },
+	{ STRCONST("addpeers"), "<peeraddr>", "add peer", do_addpeers },
+	{ STRCONST("droppeers"), "", "disconnect all peers", do_droppeers },
+	{ STRCONST("exit"), "", "terminate p2pool", do_exit },
+	{ STRCNULL, NULL, NULL, NULL }
+};
+
+static int do_help(p2pool *m_pool, const char *args)
+{
+	int i;
+
+	LOGINFO(0, "List of commands");
+	for (i=0; cmds[i].name.len; i++)
+		LOGINFO(0, cmds[i].name.str << " " << cmds[i].arg << "\t" << cmds[i].descr);
+	(void) m_pool;
+	(void) args;
+	return 0;
+}
+
+static int do_status(p2pool *m_pool, const char *args)
+{
+	m_pool->side_chain().print_status();
+	if (m_pool->stratum_server()) {
+		m_pool->stratum_server()->print_status();
+	}
+	if (m_pool->p2p_server()) {
+		m_pool->p2p_server()->print_status();
+	}
+	(void) args;
```

Comment:
I prefer to comment out the parameter name: `do_status(p2pool *m_pool, const char * /*args*/)` from the function declaraction instead of silencing the compiler like this.

Reply:
Done